test/main/utilities.js

```diff
@@ -0,0 +1,43 @@
+QUnit.module( "QUnit.objectType" );
+
+function buildFunctions() {
+	var objects = [ function( ) { } ];
+
+	try {
+		var arrow = new Function( "return () => {};" )( );
+		objects.push( arrow );
+	} catch ( e ) { }
+
+	try {
+		var async = new Function( "return async function( ) { };" )( );
+		objects.push( async );
+	} catch ( e ) { }
```

Comment:
why this try / catch + wrapping the function forms in the Function ctor?  ES6 compat?

This might escalate in a weird way, as we might also have method definitions, generators, classes, async generators...